src/toolkits/object_detection/object_detector.cpp

```diff
@@ -959,6 +959,15 @@ std::shared_ptr<MLModelWrapper> object_detector::export_to_coreml(
 std::unique_ptr<data_iterator> object_detector::create_iterator(
     gl_sframe data, std::vector<std::string> class_labels, bool repeat) const
 {
+  // Check annotations type. If it is a single dictionary, put it into a list.
+  const std::string& annotations_column_name =
+      read_state<flex_string>("annotations");
+  if (data.contains_column(annotations_column_name)) {
+    gl_sarray annotations = data[annotations_column_name];
+    data.replace_add_column(canonicalize_annotations(annotations),
+                            annotations_column_name);
+  }
+
```

Comment:
I think logic might be better suited to live in `od_data_iterator.cpp`, which is where we currently reason about the data format.